cmd/kubeadm/app/cmd/phases/init/certs.go

```diff
@@ -197,6 +199,33 @@ func runCerts(c workflow.RunData) error {
 	}
 
 	fmt.Printf("[certs] Using certificateDir folder %q\n", data.CertificateWriteDir())
+
+	// If using an external CA while dryrun, need to copy CA files to dryrun dir for later use
+	if data.ExternalCA() && data.DryRun() {
+		cfg := data.Cfg()
+		externalCAFiles := []string{
+			filepath.Join(cfg.CertificatesDir, kubeadmconstants.CACertName),
+			filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.AdminKubeConfigFileName),
+			filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.KubeletKubeConfigFileName),
+			filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.ControllerManagerKubeConfigFileName),
+			filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.SchedulerKubeConfigFileName),
```

Comment:
ok, so thinking further here.
kubeadm has 2 phases that are separate for certs and kubeconfig.

thus, i think we can split the list of filesL
- manage the CA cert in the certs phase
- manage the kubeconfig files in the kubeconfig phase

accordingly:
- we should update the comment on top of the block depending on what is done (CA cert vs kubeconfig files)
- we can also remove the detailed fmt.Printf(...) bellow per file, this feels like a "patch" on our side that the user doesn't need to necessarily know about?

Reply:
You are absolutely right! I should consider more about the phases, thanks.
done